src/util/obversion.rs

```diff
@@ -0,0 +1,131 @@
+/*-
+ * #%L
+ * OBKV Table Client Framework
+ * %%
+ * Copyright (C) 2023 OceanBase
+ * %%
+ * OBKV Table Client Framework is licensed under Mulan PSL v2.
+ * You can use this software according to the terms and conditions of the
+ * Mulan PSL v2. You may obtain a copy of Mulan PSL v2 at:
+ *          http://license.coscl.org.cn/MulanPSL2
+ * THIS SOFTWARE IS PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES OF ANY
+ * KIND, EITHER EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * NON-INFRINGEMENT, MERCHANTABILITY OR FIT FOR A PARTICULAR PURPOSE.
+ * See the Mulan PSL v2 for more details.
+ * #L%
+ */
+
+use std::sync::atomic::{AtomicU64, Ordering::Relaxed};
+
+lazy_static! {
+    pub static ref OB_VERSION: AtomicU64 = AtomicU64::new(0);
+}
+
+#[allow(dead_code)]
+const OB_VSN_MAJOR_SHIFT: u64 = 32;
+#[allow(dead_code)]
+const OB_VSN_MINOR_SHIFT: u64 = 16;
+#[allow(dead_code)]
+const OB_VSN_MAJOR_PATCH_SHIFT: u64 = 8;
+#[allow(dead_code)]
+const OB_VSN_MINOR_PATCH_SHIFT: u64 = 0;
+#[allow(dead_code)]
+const OB_VSN_MAJOR_MASK: u64 = 0xffffffff;
+#[allow(dead_code)]
+const OB_VSN_MINOR_MASK: u64 = 0xffff;
+#[allow(dead_code)]
+const OB_VSN_MAJOR_PATCH_MASK: u64 = 0xff;
+#[allow(dead_code)]
+const OB_VSN_MINOR_PATCH_MASK: u64 = 0xff;
+
+#[allow(dead_code)]
+pub fn calc_version(major: i32, minor: i16, major_patch: i8, minor_patch: i8) -> u64 {
+    ((major as u64) << OB_VSN_MAJOR_SHIFT)
+        + ((minor as u64) << OB_VSN_MINOR_SHIFT)
+        + ((major_patch as u64) << OB_VSN_MAJOR_PATCH_SHIFT)
+        + ((minor_patch as u64) << OB_VSN_MINOR_PATCH_SHIFT)
+}
```

Comment:
It seems we can use '|' to replace '+' here?

Reply:
Fixed